doc/guides/modules.rst

```diff
@@ -188,11 +129,108 @@ section.
    zephyr-modules.cmake`` to your CMake command line.
 
 Not Using Modules
-*****************
+=================
 
 If you don't have west installed and don't specify :makevar:`ZEPHYR_MODULES`
 yourself, then no additional modules are added to the build. You will still be
 able to build any applications that don't require code or Kconfig options
 defined in an external repository.
 
+.. _submitting_new_modules:
+
+Submitting a new Module
+***********************
+
+Requirements
+============
+
+Modules to be included in the default manifest of the Zephyr project need to
+provide functionality or features endorsed and approved by the project technical
+steering committee and should follow the project licencing and contribution
+guidelines.
+
+A request for a new module should be initiated using an RFC issue in the
+Zephyr project issue tracking system with details about the module and how it
+integrates into the project. If the request is approved, a new repository is
+created and initiated with basic information that would allow submitting code to
+the module project following the project contribution guidelines.
+
+Process
+=======
+
+Follow the following steps to request a new module:
+
+#. Submit an issue with details about the module to be created
+#. propose a name for the repository to be created under the Zephyr project
+   organization on Github.
+#. Maintainers from the Zephyr project will create the repository and initialize
+   it. You will be added as a collaborator in the new repository.
+#. Submit the module content (Code) to the new repository following the
+   guidelines described :ref:`here <modules_using_west>`.
+#. Submit a new entry into the :zephyr_file:`west.yml` with the following
+   information:
+
+   .. code-block:: console
+
+        - name: <name of repository>
+          path: <path to where the repository should be cloned>
+          revision: <ref pointer to module pull request>
+
+
+For example, to add *my_module* to the manifest:
+
+.. code-block:: console
+
+    - name: my_module
+      path: modules/lib/my_module
+      revision: pull/23/head
+
+
+Where 23 in the example above indicated the pull request submitted to the
+*my_module* repository.
```

Comment:
I think it's worth noting here that requesting the pull head is for CI to be able to test it out. Then mention we'll merge the module code PR and then they need to update the west.yml PR with the final revision before we can merge that, making the module an official part of the project.